m3/src/main/java/com/uber/m3/tally/m3/MetricType.java

```diff
@@ -18,15 +18,13 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package com.uber.m3.tally;
+package com.uber.m3.tally.m3;
 
 /**
- * A cached gauge for reporting an individual {@link Gauge}.
+ * Enum of the type of metric.
  */
-public interface CachedGauge {
-    /**
-     * Reports a value for a {@link Gauge}.
-     * @param value value to report
-     */
-    void reportGauge(double value);
+public enum MetricType {
+    COUNTER,
+    GAUGE,
+    TIMER,
```

Comment:
Don't need a histogram?

Reply:
I have sincec removed `MetricType` after removing the `Cached*` usages. Previously, it was not needed because histogram functionality is actually built on top of counters.

Reply:
Sounds good